### PR TITLE
Add cdi-add-interceptor-binding quickstart, JDF-154

### PR DESCRIPTION
**Pull request by pmuir**

### CODE REVIEW

File: cdi-add-interceptor-binding/src/main/java/org/jboss/as/quickstart/cdi/parameterlogger/ParameterLoggerExtension.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.apache.deltaspike.core.util.metadata.AnnotationInstanceProvider;
+import org.apache.deltaspike.core.util.metadata.builder.AnnotatedTypeBuilder;
+import org.jboss.as.quickstart.cdi.parameterlogger.model.LogParameterValue;
+import org.jboss.as.quickstart.cdi.parameterlogger.model.MyBean;
+import org.jboss.as.quickstart.cdi.parameterlogger.model.ParameterLogger;
+
+/**
+ * A simple CDI Portable Extension to add a interceptor binding annotation if certain parameter annotations are present
+ */
+public class ParameterLoggerExtension implements Extension {
```
Comment (LightGuard) [inline cdi-add-interceptor-binding/src/main/java/org/jboss/as/quickstart/cdi/parameterlogger/ParameterLoggerExtension.java]: It should probably be noted (or better, enhanced to use) this doesn't use the deactivation utility in DeltaSpike to "turn off" an extension. All of the extensions in DeltaSpike, and hopefully those built using DeltaSpike, will use ability so developers can disable an extension in their application if they do not want it to run.

**Comment by sgilda** [comment]
Except for a few minor comments in the README file, this looks good, runs great!
One question: Do all the java files need a copyright notice in the header?

**Comment by LightGuard** [comment]
@sgilda I think that's been the way we've been doing it. According to Richard it's somewhat of a grey area and in his opinion not really necessary.

**Comment by sgilda** [comment]
@LightGuard : I only asked because I remember @pmuir running something that added the copyright comments to the beginning of all the files before we shipped EAP 6. It was a massive checkin. :-)

**Comment by LightGuard** [comment]
Oh, that's right. We could always run it again before our next massive ship.

On Mon, Nov 26, 2012 at 1:20 PM, sgilda notifications@github.com wrote:

> @LightGuard https://github.com/LightGuard : I only asked because I
> remember @pmuir https://github.com/pmuir running something that added
> the copyright comments to the beginning of all the files before we shipped
> EAP 6. It was a massive checkin. :-)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jboss-jdf/jboss-as-quickstart/pull/360#issuecomment-10732028.

## 

Jason Porter
http://lightguard-jp.blogspot.com
http://twitter.com/lightguardjp

Software Engineer
Open Source Advocate

PGP key id: 926CCFF5
PGP key available at: keyserver.net, pgp.mit.edu

**Comment by sgilda** [comment]
True, but it's easier to catch them as we add them. As I recall, we ran into a few issues ;-).

**Comment by LightGuard** [comment]
Good point

On Mon, Nov 26, 2012 at 2:26 PM, sgilda notifications@github.com wrote:

> True, but it's easier to catch them as we add them. As I recall, we ran
> into a few issues ;-).
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jboss-jdf/jboss-as-quickstart/pull/360#issuecomment-10734618.

## 

Jason Porter
http://lightguard-jp.blogspot.com
http://twitter.com/lightguardjp

Software Engineer
Open Source Advocate

PGP key id: 926CCFF5
PGP key available at: keyserver.net, pgp.mit.edu

**Comment by pmuir** [comment]
Adressed comments :-)

**Comment by sgilda** [comment]
@pmuir: Were you going to combine the information under "Run tests from JBDS" section with the "Run the Quickstart in JBoss Developer Studio or Eclipse" down below? It seems like they are duplicating some of the information.

**Comment by pmuir** [comment]
Thanks @sgilda, I've done that.

**Comment by sgilda** [comment]
Merged.

**Pull request closed without merging**